src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Literal.cs

```diff
@@ -38,13 +38,22 @@ private BoundExpression MakeLiteral(SyntaxNode syntax, ConstantValue constantVal
                 Debug.Assert(type is { SpecialType: SpecialType.System_DateTime });
                 return MakeDateTimeLiteral(syntax, constantValue);
             }
-            else if (oldNodeOpt != null)
-            {
-                return oldNodeOpt.Update(constantValue, type);
-            }
             else
             {
-                return new BoundLiteral(syntax, constantValue, type, hasErrors: constantValue.IsBad);
+                if (constantValue.IsString && constantValue.StringValue.Length > _compilation.DataSectionStringLiteralThreshold)
```

Comment:
>if (constantValue.IsString && constantValue.StringValue.Length > _compilation.DataSectionStringLiteralThreshold) [](http://example.com/codeflow?start=16&length=112)

I do not think this change addresses the issue. Not all literals go through this function and they do not have to doe that.

Reply:
Okay, thanks. Moved. I'm not sure if it would be better to create a new CompletionPart for it.